File: ch5/5.4/ObjC/CollectionViewSample/CollectionViewSample/EventCollectionViewCell.m
//
//  EventCollectionViewCell.m
//  CollectionViewSample
//
//  Created by tonyguan on 2016/10/23.
//  Copyright © 2016年 关东升. All rights reserved.
//

#import "EventCollectionViewCell.h"

@implementation EventCollectionViewCell

- (id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        
        //单元格的宽度
        CGFloat cellWidth = self.frame.size.width;
        
        CGFloat imageViewWidth = 101;
        CGFloat imageViewHeight = 101;
        CGFloat imageViewTopView = 15;
        
        /// 1.添加ImageView
        self.imageView = [[UIImageView alloc] initWithFrame: CGRectMake((cellWidth - imageViewWidth) / 2, imageViewTopView, imageViewWidth, imageViewHeight)];
        
        [self addSubview:self.imageView];
        
        /// 2.添加标签
        CGFloat labelWidth = 101;
        CGFloat labelHeight = 16;
        CGFloat labelTopView = 120;
        self.label = [[UILabel alloc] initWithFrame:CGRectMake((cellWidth - labelWidth) / 2, labelTopView, labelWidth, labelHeight)];
        //字体左右居中
        self.label.textAlignment = NSTextAlignmentCenter;
        //设置字体
        self.label.font = [UIFont systemFontOfSize:13];
        [self addSubview:self.label];
        
    }
    return self;
}

@end

File: ch5/5.4/Swift/CollectionViewSample/CollectionViewSample/EventCollectionViewCell.swift
//
//  EventCollectionViewCell.swift
//  CollectionViewSample
//
//  Created by tonyguan on 2016/10/23.
//  Copyright © 2016年 关东升. All rights reserved.
//

import UIKit

class EventCollectionViewCell: UICollectionViewCell {
    
    var imageView: UIImageView!
    var label: UILabel!
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        /// 1.添加ImageView
        let cellWidth: CGFloat = self.frame.size.width
        
        let imageViewWidth: CGFloat = 101
        let imageViewHeight: CGFloat = 101
        let imageViewTopView: CGFloat = 15
        
        /// 1.添加ImageView
        self.imageView = UIImageView(frame: CGRect(x: (cellWidth - imageViewWidth) / 2, y: imageViewTopView, width: imageViewWidth, height: imageViewHeight))
        
        self.addSubview(self.imageView)
        
        /// 2.添加标签
        let labelWidth: CGFloat = 101
        let labelHeight: CGFloat = 16
        let labelTopView: CGFloat = 120
        
        self.label = UILabel(frame: CGRect(x: (cellWidth - labelWidth) / 2, y: labelTopView, width: labelWidth, height: labelHeight))
        
        //字体左右居中
        self.label.textAlignment = .center
        //设置字体
        self.label.font = UIFont.systemFont(ofSize: 13)
        self.addSubview(self.label)
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
